### ISSUE TITLE
_LocalizationsState return empty Container()

### ISSUE THREAD

**Issue by kaparray** [flutter/flutter#58502]
Why in _LocalizationsState() if _locale equals ```null``` return Empty Container()?
May be throw Exception or make assertion?

This is realisation of _LocalizationsState().
```dart
@override
  Widget build(BuildContext context) {
    if (_locale == null)
      return Container();
    return Semantics(
      textDirection: _textDirection,
      child: _LocalizationsScope(
        key: _localizedResourcesScopeKey,
        locale: _locale,
        localizationsState: this,
        typeToResources: _typeToResources,
        child: Directionality(
          textDirection: _textDirection,
          child: widget.child,
        ),
      ),
    );
  }
```

```shell
flutter doctor -v
[✓] Flutter (Channel stable, v1.17.2, on Mac OS X 10.15.3 19D76, locale en-RU)
    • Flutter version 1.17.2 at /Users/kirilladeshchenko/development/flutter
    • Framework revision 5f21edf8b6 (5 days ago), 2020-05-28 12:44:12 -0700
    • Engine revision b851c71829
    • Dart version 2.8.3
```

**Comment by goderbauer** [comment]
This is by design because the locale may be loading asynchronously and become available later, see the code here:

https://github.com/flutter/flutter/blob/3e63411256cc88afc48044aa5ea06c5c9c6a6846/packages/flutter/lib/src/widgets/localizations.dart#L501

Therefore, we can't throw or assert. In practice, the empty container will not be seen by the user because we delay rendering the first frame until the locale has loaded (see the call to deferFirstFrame in the code linked above).

I am closing this for now since it is not describing a bug or feature request. If you have any concerns about the current behavior, please describe what the problem is that you are experiencing with this implementation.

**Comment by kaparray** [comment]
@goderbauer if my custom load() method in delegate throw exceptions, then Container visible always 


```
Launching lib/main.dart on iPhone SE (2nd generation) in debug mode...
Running Xcode build...

 └─Compiling, linking and signing...                         6.6s
Xcode build done.                                           20.3s
[VERBOSE-2:ui_dart_state.cc(157)] Unhandled Exception: Exception
#0      AppLocalization.load (package:tennisi/res/app_localization.dart:41:5)
#1      _AppLocalizationDelegate.load (package:tennisi/res/app_localization.dart:19:36)
#2      _loadAll (package:flutter/src/widgets/localizations.dart:58:49)
#3      _LocalizationsState.load (package:flutter/src/widgets/localizations.dart:509:62)
#4      _LocalizationsState.initState (package:flutter/src/widgets/localizations.dart:474:5)
#5      StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:4640:58)
#6      ComponentElement.mount (package:flutter/src/widgets/framework.dart:4476:5)
#7      Element.inflateWidget (package:flutter/src/widgets/framework.dart:3446:14)
#8      Element.updateChild (package:flutter/src/widgets/framework.dart:3214:18)
#9      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4527:16)
#10     Element.rebuild (package:flutter/src/widgets/framework.dart:4218:5)
#11     ComponentElement.<…>
Syncing files to device iPhone SE (2nd generation)...              235ms

Flutter run key commands.
r Hot reload. 🔥🔥🔥
R Hot restart.
h Repeat this help message.
d Detach (terminate "flutter run" but leave application running).
c Clear the screen
q Quit (terminate the application on the device).
An Observatory debugger and profiler on iPhone SE (2nd generation) is available at: http://127.0.0.1:56427/-32OnjH565E=/


flutter: WidgetsFlutterBinding - CHECKED MODE
flutter: [root](renderObject: RenderView#4be9c)
flutter: └App(state: _AppState#a8cac)
flutter:  └StreamBuilder<dynamic>(state: _StreamBuilderBaseState<dynamic, AsyncSnapshot<dynamic>>#cf4bf)
flutter:   └MaterialApp(state: _MaterialAppState#6a20e)
flutter:    └ScrollConfiguration(behavior: _MaterialScrollBehavior)
flutter:     └WidgetsApp-[GlobalObjectKey _MaterialAppState#6a20e](state: _WidgetsAppState#67697)
flutter:      └Shortcuts(shortcuts: <Default WidgetsApp Shortcuts>, state: _ShortcutsState#dc719)
flutter:       └Focus(debugLabel: "Shortcuts", state: _FocusState#fa76b)
flutter:        └_FocusMarker
flutter:         └Semantics(container: false, properties: SemanticsProperties, label: null, value: null, hint: null, hintOverrides: null, renderObject: RenderSemanticsAnnotations#73fd8)
flutter:          └_ShortcutsMarker
flutter:           └Actions(dispatcher: null, actions: {[<DoNothingAction>]: Closure: () => DoNothingAction, [<RequestFocusAction>]: Closure: () => RequestFocusAction, [<NextFocusAction>]: Closure: () => NextFocusAction, [<PreviousFocusAction>]: Closure: () => PreviousFocusAction, [<DirectionalFocusAction>]: Closure: () => DirectionalFocusAction, [<ScrollAction>]: Closure: () => ScrollAction})
flutter:            └FocusTraversalGroup(policy: ReadingOrderTraversalPolicy#8483f, state: _FocusTraversalGroupState#dafdb)
flutter:             └_FocusTraversalGroupMarker
flutter:              └Focus(node: FocusNode#3ba2b(FocusTraversalGroup), dependencies: [_FocusMarker], state: _FocusState#32a4c)
flutter:               └_FocusMarker
flutter:                └_MediaQueryFromWindow(state: _MediaQueryFromWindowsState#32a29)
flutter:                 └MediaQuery(MediaQueryData(size: Size(375.0, 667.0), devicePixelRatio: 2.0, textScaleFactor: 1.0, platformBrightness: Brightness.light, padding: EdgeInsets(0.0, 20.0, 0.0, 0.0), viewPadding: EdgeInsets(0.0, 20.0, 0.0, 0.0), viewInsets: EdgeInsets.zero, physicalDepth: 1.7976931348623157e+308, alwaysUse24HourFormat: false, accessibleNavigation: false, highContrast: false, disableAnimations: false, invertColors: false, boldText: false))
flutter:                  └Localizations(locale: en, delegates: [AppLocalization.delegate(2 locales), GlobalMaterialLocalizations.delegate(77 locales), GlobalWidgetsLocalizations.delegate(all locales), GlobalCupertinoLocalizations.delegate(76 locales), DefaultMaterialLocalizations.delegate(en_US), DefaultCupertinoLocalizations.delegate(en_US), DefaultWidgetsLocalizations.delegate(en_US)], state: _LocalizationsState#738aa)
flutter:                   └Container
flutter:                    └LimitedBox(maxWidth: 0.0, maxHeight: 0.0, renderObject: RenderLimitedBox#7ae0f)
flutter:                     └ConstrainedBox(BoxConstraints(biggest), renderObject: RenderConstrainedBox#f13fe)
```
